systems/subscriber/sim-pool/cmd/server/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"os"
+
+	"github.com/gofrs/uuid"
+	"github.com/ukama/ukama/systems/common/msgBusServiceClient"
+
+	"github.com/num30/config"
+	pkg "github.com/ukama/ukama/systems/subscriber/sim-pool/pkg"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/pkg/server"
+	"gopkg.in/yaml.v3"
+
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	ccmd "github.com/ukama/ukama/systems/common/cmd"
+	ugrpc "github.com/ukama/ukama/systems/common/grpc"
+	mb "github.com/ukama/ukama/systems/common/msgBusServiceClient"
+	egenerated "github.com/ukama/ukama/systems/common/pb/gen/events"
+	"github.com/ukama/ukama/systems/common/sql"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/cmd/version"
+	pb "github.com/ukama/ukama/systems/subscriber/sim-pool/pb/gen"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/pkg/db"
+
+	"google.golang.org/grpc"
+)
+
+ var serviceConfig *pkg.Config
+
+
+func main() {
+	ccmd.ProcessVersionArgument(pkg.ServiceName, os.Args, version.Version)
+	pkg.InstanceId = os.Getenv("POD_NAME")
+
+	initConfig()
+	simDb := initDb()
+	runGrpcServer(simDb)
+}
+
+func initConfig() {
+	serviceConfig = pkg.NewConfig(pkg.ServiceName)
+	err := config.NewConfReader(pkg.ServiceName).Read(serviceConfig)
+	if err != nil {
+		log.Fatal("Error reading config ", err)
+	} else if serviceConfig.DebugMode {
+		b, err := yaml.Marshal(serviceConfig)
+		if err != nil {
+			logrus.Infof("Config:\n%s", string(b))
+		}
+	}
+
+	pkg.IsDebugMode = serviceConfig.DebugMode
+}
+
+func initDb() sql.Db {
+	log.Infof("Initializing Database")
+	d := sql.NewDb(serviceConfig.DB, serviceConfig.DebugMode)
+	err := d.Init(&db.Sim{})
+
+	if err != nil {
+		log.Fatalf("Database initialization failed. Error: %v", err)
+	}
+	return d
+}
+
+func runGrpcServer(gormdb sql.Db) {
+	
+	instanceId := os.Getenv("POD_NAME")
+
+	if instanceId == "" {
+		inst, err := uuid.NewV4()
+		if err != nil {
+			log.Fatalf("Failed to genrate instanceId. Error %s", err.Error())
+		}
+		instanceId = inst.String()
+	}
+
+	// timeout := serviceConfig.MsgClient.Timeout
+	if instanceId == "" {
```

Comment:
Repeated code from line 70

Reply:
I have deleted and used the previous instantId from pkg